### PR TITLE
[Zero-Dim] Support input 0D Tensor for masked_select

### PR DESCRIPTION
**Pull request by ZzSean**
### PR types

New features
### PR changes

APIs
### Describe

masked_select支持0D tensor

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 1]
```diff
@@ -1173,6 +1173,17 @@ def test_atan2(self):
...
 
+    def test_maseked_select(self):
+        x = paddle.rand([])
+        x.stop_gradient = False
+        mask = paddle.full([], True, dtype='bool')
+        y = paddle.masked_select(x, mask)
+
+        y.backward()
+
+        self.assertEqual(y.shape, [1])
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:1309]: 加下y的具体值的测试吧，与x相等

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 2]
```diff
@@ -1173,6 +1173,17 @@ def test_atan2(self):
...
+        self.assertEqual(x.grad.shape, [])
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:1312]: 加下x.grad的具体值的测试吧

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 3]
```diff
@@ -1173,6 +1173,17 @@ def test_atan2(self):
         self.assertEqual(x1.grad.numpy(), 0.5)
         self.assertEqual(x2.grad.numpy(), 0)
 
+    def test_maseked_select(self):
+        x = paddle.rand([])
+        x.stop_gradient = False
+        mask = paddle.full([], True, dtype='bool')
+        y = paddle.masked_select(x, mask)
+
+        y.backward()
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:1308]: 拉下最新的，需要y.retain_grads()

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 2 above)
Comment (ZzSean) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:1312]: done

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 1 above)
Comment (ZzSean) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:1309]: done

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 3 above)
Comment (ZzSean) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:1308]: done,thx

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 4]
```diff
@@ -1968,6 +1982,16 @@ def test_atan2(self):
 
         self.assertEqual(res[0].shape, ())
 
+    @prog_scope()
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:1985]: 静态图也要测反向呢，paddle.static.append_backward(out.sum())

Comment (ZzSean) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:1985]: done, thx

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Pull request merged**